wagtail_streamfield_migration_toolkit/management/commands/streamdatamigration.py

```diff
@@ -0,0 +1,230 @@
+        # alter_value_parser.add_argument("app_label", help="app name")
+        # alter_value_parser.add_argument("new_value", help="new value")
+        # alter_value_parser.add_argument(
+        #     "paths",
+        #     nargs="+",
+        #     action="extend",
+        #     help="path/s to the block/s which is being operated on: '<model_name>.<field_name>.?<block_path>'",
+        # )
+
+        # TODO support for more operations : TODO alter value
+
+        # TODO determine how to go ahead with multiple blocks AND operations
+
+    def handle(self, *args, **options):
+
+        self.operation_maker = {
+            "rename": self.make_rename_operation,
+            "remove": self.make_remove_operation,
+            # "alter": None,  # TODO
+        }[options["operation_type"]]
+        self.paths = options["paths"]
+        self.migration_name = options["name"]
+        self.app_label = options["app_label"]
+        self._migration_names = []
+
+        # get the project state
+        loader = MigrationLoader(connection=connection)
+        loader.build_graph()
+        self.project_state = loader.project_state()
+
+        # Since a single `MigrateStreamData` operation is defined for a specific streamfield,
+        # we will keep the intra field operations for each specific streamfield here.
+        operations_and_block_paths_by_model_field = {}
+
+        for path in self.paths:
+            model_name, field_name, block_path = self.parse_path(path)
+            operation_and_block_path = self.operation_maker(
+                block_path=block_path,
+                model_name=model_name,
+                field_name=field_name,
+                **options
+            )
+            key = (model_name, field_name)
+            if key in operations_and_block_paths_by_model_field:
+                operations_and_block_paths_by_model_field[key].append(
+                    operation_and_block_path
+                )
+            else:
+                operations_and_block_paths_by_model_field[key] = [
+                    operation_and_block_path
+                ]
+
+        migration = Migration(self.migration_name, self.app_label)
+        migration.operations = []
+        for (
+            model_name,
+            field_name,
+        ), operations_and_block_paths in (
+            operations_and_block_paths_by_model_field.items()
+        ):
+            migration.operations.append(
+                MigrateStreamData(
+                    app_name=self.app_label,
+                    model_name=model_name,
+                    field_name=field_name,
+                    operations_and_block_paths=operations_and_block_paths,
+                )
+            )
+
+        # If the user doesn't give a name for the migration file, generate one based on the
+        # operations used.
+        if not self.migration_name:
+            self.migration_name = "_".join(self._migration_names)[:30]
```

Comment:
`Migration` has a `suggest_name` method. Could we use that here? The entrypoint from `makemigrations` is here: https://github.com/django/django/blob/main/django/db/migrations/autodetector.py#L1679

Reply:
This uses a property which is not available on `RunPython` class. If we use it we will have to define the method in our `MigrateStreamData` class like here https://github.com/django/django/blob/f3cd252cfc46c0c7d66e765818dd3dadf60d4d0e/django/db/migrations/operations/fields.py#L127 .  It feels like it'll be easier to do it here for now?